overlord/snapstate/snapstate.go

```diff
@@ -47,9 +47,13 @@ const (
 	// always be enforcing, even if the snap requests otherwise.
 	JailMode
 
-	// if we need flags for just SnapSetup it may be easier
-	// to start a new sequence from the other end with:
-	// 0x40000000 >> iota
+	// flags just for SnapSetup follow
+
+	// HasABA is a feature flag; doInstall sets it, and downstream handlers check for it to handle old SnapSetups
+	HasABA = 0x40000000 >> iota
```

Comment:
yes, using a permanent feature flag in SS seems strange, a migration would be easier on this code, not sure how the migration itself would look like though (it seems the old code would just give up on ABA so that's not a case there, but we have to know revert vs not and different level of completion through the chain? )

also afaict the code to deal with the old changes seems a bit broken (see comment below about had-candidate)